utils/src/main/java/io/helidon/build/util/MavenVersions.java

```diff
@@ -0,0 +1,261 @@
+
+        /**
+         * The Maven central repository URI; used as the default if not specified.
+         */
+        public static final URI MAVEN_CENTRAL_URI = toUri("https://repo.maven.apache.org/maven2/");
+
+        private URI repositoryBaseUri;
+        private String artifactGroupId;
+        private String artifactId;
+        private String source;
+        private List<MavenVersion> versions;
+        private List<String> fallbackVersions;
+        private Predicate<MavenVersion> filter;
+
+        private Builder() {
+            this.repositoryBaseUri = MAVEN_CENTRAL_URI;
+            this.filter = v -> true;
+        }
+
+        /**
+         * Sets the repository base URI.
+         *
+         * @param repositoryUri The base URI, e.g. {@link #MAVEN_CENTRAL_URI}.
+         * @return This instance.
+         */
+        public Builder repository(URI repositoryUri) {
+            this.repositoryBaseUri = requireNonNull(repositoryUri);
+            return this;
+        }
+
+        /**
+         * Sets the artifact group id.
+         *
+         * @param artifactGroupId The id.
+         * @return This instance.
+         */
+        public Builder artifactGroupId(String artifactGroupId) {
+            this.artifactGroupId = requireNonNull(artifactGroupId);
+            return this;
+        }
+
+        /**
+         * Sets the artifact id.
+         *
+         * @param artifactId The id.
+         * @return This instance.
+         */
+        public Builder artifactId(String artifactId) {
+            this.artifactId = requireNonNull(artifactId);
+            return this;
+        }
+
+        /**
+         * Sets a filter for the versions to include.
+         *
+         * @param filter The filter.
+         * @return This instance.
+         */
+        public Builder filter(Predicate<MavenVersion> filter) {
+            this.filter = requireNonNull(filter);
+            return this;
+        }
+
+        /**
+         * Sets versions to use as a fallback if the metadata file cannot be accessed.
+         *
+         * @param fallbackVersions The versions.
+         * @return This instance.
+         */
+        public Builder fallbackVersions(List<String> fallbackVersions) {
+            if (requireNonNull(fallbackVersions).isEmpty()) {
+                throw new IllegalArgumentException("fallback versions empty");
+            }
+            this.fallbackVersions = fallbackVersions;
+            return this;
+        }
+
+        /**
+         * Builds the {@link MavenVersions} instance.
+         *
+         * @return The instance.
+         * @throws IllegalStateException If there are no versions available.
+         */
+        public MavenVersions build() {
+            requireNonNull(artifactGroupId, "artifactGroupId is required");
+            requireNonNull(artifactId, "artifactId is required");
+            try {
+                final boolean local = repositoryBaseUri.getScheme().equals(LOCAL_URI_SCHEME);
```

Comment:
The assumption of the repository scheme based on the URI is incorrect, a remote repository can be specified using a file URI.

Also I don't understand the need to support the local repository, maybe instead you can just make the name of the metadata file configureable.